src/addons/MountNode/lib/computeClasses.js

```diff
@@ -0,0 +1,11 @@
+import _ from 'lodash/fp'
+
+const computeClasses = (components) => components && _.flow(
+  _.map('props.className'),
+  _.filter(_.isString),
+  _.map(_.split(/\s+/)),
+  _.flatten,
+  _.uniq,
+)([...components])
```

Comment:
I didn't test this, but I'm pretty sure:
- Lodash will gracefully handle null/undefined args (e.g. `components`) so no need to check if it exists first
- Lodash will not mutate the original array so no need copy if first
- There is a `_.flatMap` that will handle mapping and flattening at the same time.
- This will currently include empty string `className`s.  However, switching the filter to an `_.identity` will filter all falsy values.

This can probably be simplified to:

```js
const computeClasses = _.flow(
  _.map('props.className'),
  _.filter(_.identity),
  _.flatMap(_.split(/\s+/)),
  _.uniq,
)
```

Which works like so:
```js
computeClasses(null)      //=> []
computeClasses(undefined) //=> []

computeClasses([
  null,
  undefined,
  {},
  { props: {} },
  { props: { className: 'foo' } },
  { props: { className: 'foo bar' } },
  { props: { className: '0' } },
  { props: { className: 'false' } },
])

// => [ 'foo', 'bar', '0', 'false' ]
```